### PR TITLE
Convert NeRF example to Keras 3.

### PR DESCRIPTION
**Pull request by hertschuh**
Note that it supports the Tensorflow backend only as it uses `tf.image` and `tf.GradientTape`.

### CODE REVIEW

File: examples/vision/md/nerf.md [hunk 1]
```diff
@@ -607,206 +607,274 @@ create_gif("images/*.png", "training.gif")
...
+![png](/img/examples/vision/nerf/nerf_11_41.png)
+    
 
 
 <div class="k-default-codeblock">
 ```
-100%|██████████| 20/20 [00:00<00:00, 57.59it/s]
+ 16/16 ━━━━━━━━━━━━━━━━━━━━ 16s 963ms/step - loss: 0.0129 - psnr: 19.0232 - val_loss: 0.0129 - val_psnr: 18.9893
+
+  0%|                                                                                                                                                                                                                                                                                                                                              | 0/20 [00:00<?, ?it/s]/var/tmp/ipykernel_49657/3697329163.py:127: DeprecationWarning: Starting with ImageIO v3 the behavior of this function will switch to that of iio.v3.imread. To keep the current behavior (and make this warning disappear) use `import imageio.v2 as imageio` or call `imageio.v2.imread` directly.
```
Comment (fchollet) [inline examples/vision/md/nerf.md:875]: Please fix (you can just fix the code and scrub the warning from the md file, no need to regenerate)

Comment (hertschuh) [reply examples/vision/md/nerf.md]: Done.

**Review by fchollet — APPROVED** [review]
LGTM, thank you!
